pkg/gadgets/run/tracer/tracer.go

```diff
@@ -376,6 +376,46 @@ func (t *Tracer) attachProgram(gadgetCtx gadgets.GadgetContext, p *ebpf.ProgramS
 		case strings.HasPrefix(p.SectionName, "kretprobe/"):
 			logger.Debugf("Attaching kretprobe %q to %q", p.Name, p.AttachTo)
 			return link.Kretprobe(p.AttachTo, prog, nil)
+		case strings.HasPrefix(p.SectionName, "uprobe/"):
+			captureHost := false
+			for _, container := range t.containers {
+				if container.Pid == 1 {
+					captureHost = true
+				}
+			}
+			if !captureHost {
+				return nil, fmt.Errorf("uprobe can only be used with --host at this moment")
+			}
+			logger.Debugf("Attaching uprobe %q to %q", p.Name, p.AttachTo)
+			parts := strings.Split(p.AttachTo, ":")
+			if len(parts) < 2 {
+				return nil, fmt.Errorf("invalid section name %q", p.AttachTo)
+			}
+			ex, err := link.OpenExecutable(parts[0])
```

Comment:
We should still use `filepath.Join(host.HostProcFs, "1/root", parts[0])` because even though this PR does not support tracing containers, it should support `ig` running in a container.

I think we should mandate an absolute path with `filepath.IsAbs()`